rewrite-maven/src/test/java/org/openrewrite/maven/AddDependencyTest.java

```diff
@@ -1699,6 +1700,156 @@ void addDependencyToAggregatingPomAndParentPomWhenAggregatingPomIsParent() {
         );
     }
 
+    @Test
+    void addDependencyWithBroaderScopeChangesExistingScopeImplicit() {
+        rewriteRun(
+          spec -> spec.recipe(new AddDependency("jakarta.annotation", "jakarta.annotation-api", "2.1.1", null, null, null, null, null, null, null, null, null)),
```

Comment:
Seeing the unit test, this just looks dangerous to me. I may be saying I want to upgrade it in the compile scope but when applied in a multi-repository environment, pulling things from provided to compile will be bad. So I don't think I'm on board with the concept of this change.